sigma/plugins/fun/xkcd.py

```diff
@@ -2,27 +2,42 @@
 import aiohttp
 import discord
 
-
 async def xkcd(cmd, message, args):
 
-    comic_no = str(random.randint(1, 1724))
+    random_comic = True
     if args:
         if not is_int(args[0]):
             await message.channel.send(None, embed=discord.Embed(title=':exclamation: Invalid number', color=0x993333))
             return
-        
+
         comic_no = args[0]
-        
-    joke_url = 'http://xkcd.com/' + comic_no + '/info.0.json'
-    async with aiohttp.ClientSession() as session:
-        async with session.get(joke_url) as data:
-            try: joke_json = await data.json()
-            except Exception:
-                await message.channel.send(None, embed=discord.Embed(title=':exclamation: Requested XKCD doesn\'t exist', color=0x993333))
-                return
+        random_comic = False
+
+    while True:
+        if not random_comic:
+            joke_url = 'http://xkcd.com/' + comic_no + '/info.0.json'
+        else:
+            joke_url = 'http://xkcd.com/info.0.json'
+
+        async with aiohttp.ClientSession() as session:
```

Comment:
This could be as simple as 
```
data = urllib.request.urlopen(url).read()
data = json.loads(data.decode('utf-8'))
```

Reply:
But you explicitly told me a while ago not to use `urllib`:

![image](https://user-images.githubusercontent.com/12093732/60774206-c96ee400-a0c5-11e9-9b5b-4b74ead32551.png)

Reply:
Hmmm then don't have stuff in those; avoid nesting as much as possible:

```
async with aiohttp.ClientSession() as session:
        async with session.get(turl) as data:
            page = await data.text()

# Do stuff with page here
```

Reply:
Fixed in df76b28.

I couldn't move everything from the nesting, because the `try-except` block must be in there, or the connection closes (second info):
![image](https://puu.sh/DQmdD/d0c258b152.png)

Reply:
Don't convert it to json in the nest. See example

Reply:
```python
        async with aiohttp.ClientSession() as session:
            async with session.get(joke_url) as data:
                json_data = data

        try: joke_json = await json_data.json()
        except Exception:
            await message.channel.send(None, embed=discord.Embed(title=':exclamation: Requested xkcd doesn\'t exist', color=0x993333))
            return```

This always throws an exception.

Reply:
What type of exception?

Reply:
All it says is "Connection closed".

![image](https://user-images.githubusercontent.com/12093732/60847518-b9bfd000-a197-11e9-8a52-92224bd6df5b.png)

![image](https://user-images.githubusercontent.com/12093732/60847536-c6442880-a197-11e9-8887-2752de306c63.png)

Reply:
That makes no sense. It should print one of the things you put into the exception.

Reply:
![image](https://user-images.githubusercontent.com/12093732/60851056-b253f300-a1a6-11e9-92ec-461ed75d7e9f.png)

Reply:
I think you're making a mountain out of a molehill.